src/firebase/models/players_in_room.js

```diff
@@ -0,0 +1,34 @@
+import { ref, onValue, onDisconnect, set, child } from "firebase/database";
+import rtdb from "../rtdb";
+
+const rooms = ref(rtdb, "rooms");
+const amOnline = ref(rtdb, ".info/connected");
+
+function createNewRandomUserRef(routeRoomID) {
+  const userId =
+    "rooms/" +
+    routeRoomID +
+    "/" +
+    Math.trunc(Math.random() * 100000).toString();
+
+  return ref(rtdb, userId);
+}
+
+export function notifyMyOnlineStatus(routeRoomID) {
+  const userRef = createNewRandomUserRef(routeRoomID);
+
+  onValue(amOnline, snapshot => {
+    if (snapshot.val()) {
+      onDisconnect(userRef).remove();
+      set(userRef, true);
+    }
+  });
+}
+
```

Comment:
Struggling to follow exactly how this works, but its purpose seems to be create a function - notifyMyOnlineStatus -which tells us whether a particular user is online or not